### PR TITLE
[ticket/14258] Add event into auth::login before session creation

### PR DESCRIPTION
**Pull request by paul999**
Add a new event in the login method just before the session create so extensions can do some additional validation during login.

PHPBB3-14258

### CODE REVIEW

File: phpBB/phpbb/auth/auth.php [hunk 1]
```diff
@@ -982,6 +983,25 @@ function login($username, $password, $autologin = false, $viewonline = 1, $admin
 				redirect($url);
 			}
 
+			/**
+			 * This event is triggered just after the user has been checked if he has a valid username/password,
```
Comment (marc1706) [inline phpBB/phpbb/auth/auth.php]: I'd prefer ```Event is triggered after checking for valid username and password, and before the actual session creation```

File: phpBB/phpbb/auth/auth.php [hunk 2]
```diff
@@ -982,6 +983,25 @@ function login($username, $password, $autologin = false, $viewonline = 1, $admin
...
+			 * but before the actual session creation.
+			 *
+			 * @event core.auth_login_before_session_create
```
Comment (marc1706) [inline phpBB/phpbb/auth/auth.php]: core.auth_login_session_create_before

File: phpBB/phpbb/auth/auth.php [hunk 3]
```diff
@@ -982,6 +983,25 @@ function login($username, $password, $autologin = false, $viewonline = 1, $admin
...
+			 * @var	array	login				Variable containing login array value
```
Comment (marc1706) [inline phpBB/phpbb/auth/auth.php]: Variable containing login array

File: phpBB/phpbb/auth/auth.php [hunk 4]
```diff
@@ -982,6 +983,25 @@ function login($username, $password, $autologin = false, $viewonline = 1, $admin
...
+			 * @var	bool	admin				Variable containing if is logging into the acp
```
Comment (marc1706) [inline phpBB/phpbb/auth/auth.php]: Boolean variable whether user is admin

File: phpBB/phpbb/auth/auth.php [hunk 5]
```diff
@@ -982,6 +983,25 @@ function login($username, $password, $autologin = false, $viewonline = 1, $admin
...
+			 * @var	string	username			Variable containing the username
```
Comment (marc1706) [inline phpBB/phpbb/auth/auth.php]: Username of user to log in

File: phpBB/phpbb/auth/auth.php (hunk 4 above)
Comment (paul999) [inline phpBB/phpbb/auth/auth.php]: It is not if user is admin, but if he is going to login into the ACP

File: phpBB/phpbb/auth/auth.php [hunk 6]
```diff
@@ -982,6 +983,25 @@ function login($username, $password, $autologin = false, $viewonline = 1, $admin
...
+			 * @var	bool	autologin			Variable containing if this is by auto login
```
Comment (marc1706) [inline phpBB/phpbb/auth/auth.php]: Boolean variable signaling whether login is triggered via auto login

File: phpBB/phpbb/auth/auth.php [hunk 7]
```diff
@@ -982,6 +983,24 @@ function login($username, $password, $autologin = false, $viewonline = 1, $admin
 				redirect($url);
 			}
 
+			/**
+			 * TEvent is triggered after checking for valid username and password, and before the actual session creation.
```
Comment (marc1706) [inline phpBB/phpbb/auth/auth.php]: TEvent? :)

**Pull request merged**